File: frameworks/C++/cutelyst/src/plaintexttest.cpp
#include "plaintexttest.h"

PlaintextTest::PlaintextTest(QObject *parent) : Controller(parent)
{

}

void PlaintextTest::plaintext(Context *c)
{
    Response *res = c->response();
    res->setBody("Hello, World!"_qba);
    res->setContentType("text/plain"_qba);
}

File: frameworks/Java/officefloor/src/woof_benchmark/src/test/java/net/officefloor/benchmark/PlaintextTest.java
package net.officefloor.benchmark;

import org.junit.Rule;
import org.junit.Test;

import net.officefloor.server.http.mock.MockHttpResponse;
import net.officefloor.server.http.mock.MockHttpServer;
import net.officefloor.woof.mock.MockWoofServerRule;

/**
 * Tests the plain text.
 */
public class PlaintextTest {

	@Rule
	public MockWoofServerRule server = new MockWoofServerRule();

	@Test
	public void validRequest() {
		MockHttpResponse response = this.server.send(MockHttpServer.mockRequest("/plaintext"));
		response.assertResponse(200, "Hello, World!", "content-type", "text/plain");
	}

}
